python/delta/tests/test_deltatable.py

```diff
@@ -791,6 +791,34 @@ def test_protocolUpgrade(self) -> None:
         with self.assertRaisesRegex(ValueError, "writerVersion"):
             dt.upgradeTableProtocol(1, {})  # type: ignore[arg-type]
 
+    def test_restoreToVersion(self) -> None:
+        self.__writeDeltaTable([('a', 1), ('b', 2)])
+        self.__overwriteDeltaTable([('a', 3), ('b', 2)],
+                                   schema=["key_new", "value_new"],
+                                   overwriteSchema='true')
+        DeltaTable.forPath(self.spark, self.tempFile).restoreToVersion(0)
+        restored = DeltaTable.forPath(self.spark, self.tempFile).toDF()
+
+        self.__checkAnswer(restored, [Row(key='a', value=1), Row(key='b', value=2)])
+
+    def test_restoreToTimestamp(self) -> None:
+        self.__writeDeltaTable([('a', 1), ('b', 2)])
+        timestampToRestore = DeltaTable.forPath(self.spark, self.tempFile) \
+            .history() \
+            .head() \
+            .timestamp \
+            .strftime('%Y-%m-%d %H:%M:%S.%f')
+
+        self.__overwriteDeltaTable([('a', 3), ('b', 2)],
+                                   schema=["key_new", "value_new"],
+                                   overwriteSchema='true')
+
```

Comment:
Add check to verify that data has changed?
`self.__checkAnswer(restored, [Row(key='a', value=3), Row(key='b', value=2)])`?

Similarly in the above test

Reply:
Both tests are updated